libtransmission/peer-mgr.c

```diff
@@ -2800,8 +2803,31 @@ struct tr_peer_stat* tr_peerMgrPeerStats(tr_torrent const* tor, int* setmeCount)
         stat->progress = peer->progress;
         stat->isUTP = tr_peerMsgsIsUtpConnection(msgs);
         stat->isEncrypted = tr_peerMsgsIsEncrypted(msgs);
-        stat->rateToPeer_KBps = toSpeedKBps(tr_peerGetPieceSpeed_Bps(peer, now_msec, TR_CLIENT_TO_PEER));
-        stat->rateToClient_KBps = toSpeedKBps(tr_peerGetPieceSpeed_Bps(peer, now_msec, TR_PEER_TO_CLIENT));
+        // The rates jump around a lot, so smooth them out. We are considering this a data collection issue rather than
+        // presentation issue, and so are smoothing here rather than within the respective UIs.
```

Comment:
- I'm not opposed to smoothing, but it's not a data collection issue. Might modify this to something like just `// smooth out transfer speeds`

- Actually it might be better to modify tr_stat to have two fields per direction to show 'smoothed' and 'instant'. That would make me feel a lot better about this overall change.